src/RouteCollectionInterface.php

```diff
@@ -20,7 +20,7 @@ public function getRoute(string $name): Route;
     /**
      * Returns routes tree array
      *
-     * @return array
+     * @return string[]|string[][]|string[][][]|string[][][][]|string[][][][][]
```

Comment:
That looks super-weird. Is there a need for such construct? What does it fix?

Reply:
I would remove it from here and create some `RoutesTreeBuilder`

Reply:
OK. Closing the pull request then.